### PR TITLE
Feat/gdl localization

### PR DESCRIPTION
**Pull request by gabrielslach**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,9 +15,10 @@ This project is currently in development and is a first draft aimed at testing t
 2. Navigate to the project folder: `cd cylander`
 3. Install dependencies: `npm install`
 4. Generate scenes for development: `npm run generate:dev:all` (this add debug capability)
-5. Start the server: `npm run start` or Run watcher: `npm run watch` (regenerate dev scenes)
-6. Generate scenes for production: `npm run generate:prod:all`
-7. Run tests: `npm run test`
+5. Generate translation files: `npm run generate:translations`
```
Comment (loydle) [inline README.md:18]: I think we should not have a separated task for translation, in my opinion this should be part of generating the scenes

File: docs/scene-structure-documentation.md [hunk 2]
```diff
@@ -53,7 +53,9 @@ As of now there are three types of actionable items:
   "name": "itemTypeImage", // (mandatory)
   "type": "image",// (mandatory)
   "label": { // (optional)
-    "content": "text"
+    "content": {
+      "en": "text"
```
Comment (loydle) [inline docs/scene-structure-documentation.md:57]: I would also add the supported languages: 

```
"fr"; "",
"nl": ""
```

File: docs/scene-structure-documentation.md [hunk 3]
```diff
@@ -79,7 +81,9 @@ As of now there are three types of actionable items:
   "name": "itemTypeHitbox", // (mandatory)
   "type": "hitbox", // (mandatory)
   "label": { // (optional)
-    "content": "text"
+    "content": {
```
Comment (loydle) [inline docs/scene-structure-documentation.md:84]: I would also add the supported languages:

```
"fr"; "",
"nl": ""
```

File: docs/scene-structure-documentation.md [hunk 4]
```diff
@@ -108,7 +112,9 @@ As of now there are three types of actionable items:
   "name": "itemTypeText", // (mandatory)
   "type": "text", // (mandatory)
   "text": {
-  "content": "Click anywhere to start!", // (mandatory)
+  "content": {
```
Comment (loydle) [inline docs/scene-structure-documentation.md:115]: I would also add the supported languages:

```
"fr"; "",
"nl": ""
```

File: scene-generator/generateTranslations.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
+const fs = require('fs');
```
Comment (loydle) [inline scene-generator/generateTranslations.js:1]: If possible I would have this logic as part of generating the scenes

File: src/js/localization.js
Comment (loydle) [inline src/js/localization.js:1]: Overall, 
I'm wondering if we can not simplify the logic to have the languages content in separeted properties in the json requierements ex: 

 ```
 "content": {
          "en": "English content",
          "fr": "French content",
          "nl": "Dutch content"
          } 
    }
```

Have a "global" language key (maybe from a UI menu) ?

Then simply set the content when needed like : 

`wateverObject.content[language]`

**Review by loydle — COMMENTED** [review]
Overall, 
I'm wondering if we can not simplify the logic to have the languages content in separeted properties in the json requierements ex: 

 ```
 "content": {
          "en": "English content",
          "fr": "French content",
          "nl": "Dutch content"
          } 
    }
```

Have a "global" language key (maybe from a UI menu) ?

Then simply set the content when needed like : 

`wateverObject.content[language]`

File: README.md
Comment (loydle) [inline README.md:1]: In my opinion we should not have a separated npm task for the translations, it should be part of the "build"/generation of scenes

**Comment by gabrielslach** [comment]
> Overall, I'm wondering if we can not simplify the logic to have the languages content in separeted properties in the json requierements ex:
> 
> ```
> "content": {
>          "en": "English content",
>          "fr": "French content",
>          "nl": "Dutch content"
>          } 
>    }
> ```
> 
> Have a "global" language key (maybe from a UI menu) ?
> 
> Then simply set the content when needed like :
> 
> `wateverObject.content[language]`

Do you mean an approach not having to dynamically generate keys for every caption/label?

**Pull request closed without merging**